docs/en/development/io_processing.md

```diff
@@ -3,7 +3,51 @@ sidebar_label: Read/Write/Delete Processing Flow
 sidebar_position: 2
 slug: /internals/io_processing
 ---
-# Read and write request processing flow
+# An introduction to the workflow of processing read and write
 
-Working in progress...
+## Workflow of processing write
 
+JuiceFS splits large files at multiple levels to improve I/O performance. See [how JuiceFS store files](../reference/how_juicefs_store_files.md). When processing write requests, JuiceFS first writes data into the client buffer and stores in chunks and slices. Chunks are isolated logical units each split into 64 MiB offset by offset consequently, and will be further split into slices if necessary. A new slice will be created if it doesn't overlap nor adjoin any existing slices, otherwise, update the affected existing slices in-place instead of creating. Slices are logical units for data persistency. When doing a flush, a slice will be first split into 4 MiB blocks consequently, and then uploaded to object storage, and eventually the metadata of the slice will be upadted. There is a one-to-one correspondence between blocks and objects. As we can see, when writing consequently, **only one** increasing slice and one final flush are needed. This maximizes the write performance of the object storage. A simple [JuiceFS benchmark](../benchmark/performance_evaluation_guide.md) as shows below consequently writes a 1 GiB file at its first stage with a 1 MiB I/O size. The following figure shows the dataflow in each component of the system.
```

Comment:
```suggestion
JuiceFS splits large files at multiple levels to improve I/O performance. See [how JuiceFS store files](../reference/how_juicefs_store_files.md). When processing write requests, JuiceFS first writes data into the client buffer and stores in chunks and slices. Chunks are successive logical units, which are split in a size of 64 MiB based on file offset Chunks are isolated from each other and will be further split into slices if necessary. A new slice will be created if it doesn't overlap nor adjoin any existing slices; otherwise,  the affected existing slices will be updated. Slices are logical units for data persistency. When doing a flush, a slice will be first split into 4 MiB blocks and then uploaded to object storage, and eventually the metadata of the slice will be updated. As there is a one-to-one correspondence between blocks and objects, we only need **one** increasing slice and one final flush when writing sequentially. This maximizes the write performance of the object storage. A simple [JuiceFS benchmark](../benchmark/performance_evaluation_guide.md) below shows sequentially writing a 1 GiB file with a 1 MiB I/O size at its first stage. The following figure shows the data flow in each component of the system.
```